File: imagegame_ui/src/services/GameHistoryService.js
import http from "../http-common"; 

class GameHistoryService {
  getAllGameHistorys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/gameHistory/gameHistorys`, searchDTO);
  }

  get(gameHistoryId) {
    return this.getRequest(`/gameHistory/${gameHistoryId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/gameHistory?field=${matchData}`, null);
  }

  addGameHistory(data) {
    return http.post("/gameHistory/addGameHistory", data);
  }

  update(data) {
  	return http.post("/gameHistory/updateGameHistory", data);
  }
  
  uploadImage(data,gameHistoryId) {
  	return http.postForm("/gameHistory/uploadImage/"+gameHistoryId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new GameHistoryService();

File: src/main/java/com/imagegame/service/GameHistoryService.java
package com.imagegame.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.imagegame.domain.GameHistory;
import com.imagegame.dto.GameHistoryDTO;
import com.imagegame.dto.GameHistorySearchDTO;
import com.imagegame.dto.GameHistoryPageDTO;
import com.imagegame.dto.GameHistoryConvertCriteriaDTO;
import com.imagegame.service.GenericService;
import com.imagegame.dto.common.RequestDTO;
import com.imagegame.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface GameHistoryService extends GenericService<GameHistory, Integer> {

	List<GameHistory> findAll();

	ResultDTO addGameHistory(GameHistoryDTO gameHistoryDTO, RequestDTO requestDTO);

	ResultDTO updateGameHistory(GameHistoryDTO gameHistoryDTO, RequestDTO requestDTO);

    Page<GameHistory> getAllGameHistorys(Pageable pageable);

    Page<GameHistory> getAllGameHistorys(Specification<GameHistory> spec, Pageable pageable);

	ResponseEntity<GameHistoryPageDTO> getGameHistorys(GameHistorySearchDTO gameHistorySearchDTO);
	
	List<GameHistoryDTO> convertGameHistorysToGameHistoryDTOs(List<GameHistory> gameHistorys, GameHistoryConvertCriteriaDTO convertCriteria);

	GameHistoryDTO getGameHistoryDTOById(Integer gameHistoryId);







}





